servers/rendering/renderer_rd/storage_rd/mesh_storage.cpp

```diff
@@ -1202,7 +1262,7 @@ void MeshStorage::_mesh_surface_generate_version_for_input_mask(Mesh::Surface::V
 		}
 		int loc = attributes[i].location;
 
-		if (loc < RS::ARRAY_COLOR) {
+		if ((loc < RS::ARRAY_COLOR) || ((loc >= PREVIOUS_LOCATION_VERTEX) && (loc <= PREVIOUS_LOCATION_TANGENT))) {
```

Comment:
Probably needs to be reviewed along with the decision to place the locations defined above. Not to mention this only works if the location numbers are contiguous.